ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -2817,3 +2817,47 @@ def test_ipahealthcheck_iparaagent_bad_serial(self, update_ra_cert_desc):
                 assert check["kw"]["msg"] == error_msg
             else:
                 assert error_reason in check["kw"]["msg"]
+
+
+class TestIpaHealthCheckSingleMaster(IntegrationTest):
+
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(
+            cls.master, setup_dns=True, extra_args=['--no-dnssec-validation']
+        )
+        tasks.install_packages(cls.master, HEALTHCHECK_PKG)
+
+    def test_ipahealthcheck_mismatching_certificates_subject(self):
+        """
+        Test if healthcheck uses cert subject base from IPA and not from
+        REALM. This prevents false-positive errors when the subject base is
+        customized.
+
+        Related: https://github.com/freeipa/freeipa-healthcheck/issues/253
+        """
+        # remove existing master installation
+        tasks.uninstall_master(self.master)
+
+        # install master with custom cert subject name
+        tasks.install_master(
+            self.master,
+            setup_dns=True,
+            extra_args=[
+                '--no-dnssec-validation',
+                '--subject-base=O=LINUX.IS.GREAT,C=EU'
+            ]
+        )
```

Comment:
Do you need the class install_master to do an installation? It then gets removed in the test and re-installed.

Reply:
My idea behind it is that when the test class is extended, `install` method will be needed anyway. 
In which case either this test will need to be modified to uninstall, or the new tests will need to include separate install, which would be inconsistent.

I can be wrong though, I can remove it, if you thing it would be better thing to do :)

Reply:
Alternatively, I can add a check if the host is configured to the `tasks` module and use a conditional uninstall.

Reply:
The thing is that depending on where the test ends up either the original install will be used or whichever previous test (or uninstalled altogether).

For this testing with custom subjects it is probably better to have a specific test class for it so nobody tries to add other generic single-server tests to it. Then you can override install_master as a no-op and install/uninstall within the test(s) directly.

Note that there is also a --ca-subject option which sets the CN value of the IPA CA which should be covered (hats-off to Flo)

Reply:
Okay, thank you for your feedback, I'll edit the tests accordingly

Reply:
@rcritten I have modified the tests accordingly (to my best understanding)